portfolio/src/main/java/com/google/sps/servlets/ContactFormServlet.java

```diff
@@ -14,34 +14,38 @@
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
+import com.google.sps.data.Message;
 
 /** Servlet that processes text. */
-@WebServlet("/contact_form")
+@WebServlet("/message")
 public final class ContactFormServlet extends HttpServlet {
 
+  private DatastoreService datastore;
+  
+  @Override
+  public void init() {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    this.datastore = datastore;
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
     // Get input from form
-    String fname = getParameter(request, "fname", "");
-    String lname = getParameter(request, "lname", "");
+    String firstName = getParameter(request, "firstName", "");
+    String lastName = getParameter(request, "lastName", "");
     String email = getParameter(request, "email", "");
     String mobile = getParameter(request, "mobile", "");
-    String message = getParameter(request, "message", "");
+    String comments = getParameter(request, "comments", "");
 
-    // // Respond with the result.
-    // response.setContentType("text/html;");
-    // response.getWriter().println(form);
+    Entity formEntity = new Entity("Message");
+    formEntity.setProperty("firstName", firstName);
+    formEntity.setProperty("lastName", lastName);
+    formEntity.setProperty("email", email);
+    formEntity.setProperty("mobile", mobile);
+    formEntity.setProperty("comments", comments);
 
-    Entity form_entity = new Entity("contact_form");
-    form_entity.setProperty("fname", fname);
-    form_entity.setProperty("lname", lname);
-    form_entity.setProperty("email", email);
-    form_entity.setProperty("mobile", mobile);
-    form_entity.setProperty("message", message);
-
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.put(form_entity);
+    this.datastore.put(formEntity);
```

Comment:
No need for `this`. `datastore.put(formEntity)` will do.

Reply:
Why isn't there a need for that line? Isn't that to store the information from the form in the database?

Reply:
oh i mean you can refer to `datastore` directly, instead of using `this.datastore`

Reply:
ohhh i see ok thanks!